Sources/PackageGraph/PackageGraphLoader.swift

```diff
@@ -150,11 +148,11 @@ private func createResolvedPackages(
         // Make sure these module names are unique in the graph.
         let dependencyModuleNames = dependencies.lazy.flatMap{ $0.modules }.map{ $0.name }
         if let duplicateModules = dependencyModuleNames.duplicates(modules.lazy.map{$0.name}) {
-            errors.append(ModuleError.duplicateModule(duplicateModules.first!))
+            engine.emit(ModuleError.duplicateModule(duplicateModules.first!))
```

Comment:
Basically the errors are added into the engine and the later one someone can chose what to do with those errors. I think you're right that addError is more semantically closer but emit is fine IMHO.